doc/ch-convert_desc.rst

```diff
@@ -0,0 +1,167 @@
+      can be revisited later IMO if needed.
+
+
+Image formats
+=============
+
+:code:`ch-convert` knows about these values of :code:`FMT`:
+
+  :code:`ch-image`
+    Internal storage for Charliecloud's unprivileged image builder (Dockerfile
+    interpreter) :code:`ch-image`.
+
+  :code:`dir`
+    Ordinary filesystem directory (i.e., not a mount point) containing an
+    unpacked image. Output directories that already exist are replaced if they
+    look like an image; otherwise, exit with an error.
+
+  :code:`docker`
+    Internal storage for Docker.
+
+  :code:`squash`
+    SquashFS filesystem archive containing the flattened image. SquashFS
+    archives are much like tar archives but are mountable, including by
+    :code:`ch-run`'s internal SquashFUSE mounting. Most systems have at least
+    the SquashFS-Tools installed which allows unpacking into a directory, just
+    like tar. Due to this greater flexibility, SquashFS is preferred to tar.
+
+    **Note:** Conversions to and from SquashFS are quite noisy due to the
+    verbosity of the underlying :code:`mksquashfs(1)` and
+    :code:`unsquashfs(1)` tools.
+
+  :code:`tar`
+    Tar archive containing the flattened image with no layer sub-archives;
+    i.e., the output of :code:`docker export` works but the output of
+    :code:`docker save` does not. Output tarballs are always gzipped and must
+    end in :code:`.tar.gz`; input tarballs can be any compression acceptable
+    to :code:`tar(1)`.
+
+All of these are local formats; :code:`ch-convert` does not know how to push
+or pull images.
+
+
+Format inference
+================
+
+:code:`ch-convert` tries to save typing by guessing formats when they are
+reasonably clear. This is done against filenames, rather than file contents,
+so the rules are the same for output descriptors that do not yet exist.
+
+Format inference is done for both :code:`IN` and :code:`OUT`. The first
+matching glob below yields the inferred format. Paths need not exist in the
+filesystem.
+
+  1. :code:`*.sqfs`, :code:`*.squash`, :code:`*.squashfs`: SquashFS.
+
+  2. :code:`*.tar`, :code:`*.t?z`, :code:`*.tar.?`, :code:`*.tar.??`: Tarball.
+
+  3. :code:`/*`, :code:`./*`, i.e. absolute path or relative path with
+     explicit dot: Directory.
+
+  4. If `ch-image` is installed: :code:`ch-image` internal storage.
+
+  5. If Docker is installed: Docker internal storage.
+
+  6. Otherwise: No format inference.
+
+
+Examples
+========
+
+Typical build-to-run sequence for image :code:`foo/bar` using :code:`ch-run`'s
+internal SquashFUSE code, inferring the output format::
+
+  $ sudo docker build -t foo/bar -f Dockerfile .
+  [...]
+  $ ch-convert foo/bar:latest /var/tmp/foobar.sqfs
+  input:   docker    foo/bar:latest
+  output:  squashfs  /var/tmp/foobar.sqfs
+  copying ...
+  done
+  $ ch-run /var/tmp/foobar.sqfs -- echo hello
+  hello
+
+Same conversion, but no format inference::
+
+  $ ch-convert -i ch-image -o squash foo/bar:latest /var/tmp/foobar.sqfs
```

Comment:
If I tell `ch-convert` to explicitly give me a squash and give it a .tar.gz extension it is totally fine with that. Not sure if maybe we want to raise a warning of a format mismatch in explicit out situations.

Reply:
Ubuntu 20.04 also has no qualms coloring my ls output making it look like it is really a tar file too.

![image](https://user-images.githubusercontent.com/8637290/138179526-e57faf1f-445d-4239-bcf3-19a6b1534e08.png)